sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -81,6 +89,143 @@ def __init__(
             )
         self.job_kind = job_kind
 
+    def train(
+        self,
+        name: str = None,
+        namespace: str = None,
+        num_workers: int = 1,
+        num_procs_per_worker: int = 1,
+        storage_config: Dict[Literal["size", "storage_class"], str] = None,
+        model_provider_parameters: HuggingFaceModelParams = None,
+        dataset_provider_parameters: Union[HfDatasetParams, S3DatasetParams] = None,
+        train_parameters: HuggingFaceTrainParams = None,
+        resources_per_worker: Dict[Literal["gpu", "cpu", "memory"], any] = None,
+    ):
+        """
+        Higher level train api
+        """
+        if (
+            not name
+            or not storage_config
+            or not model_provider_parameters
+            or not dataset_provider_parameters
+            or not train_parameters
+            or not resources_per_worker
+        ):
+            raise ValueError("One of the required parameters is None")
+
+        namespace = namespace or self.namespace
+
+        if "cpu" not in resources_per_worker or "memory" not in resources_per_worker:
+            raise ValueError("cpu and memory resources not specified")
```

Comment:
@deepanker13 Why user has to set the CPU and Memory resources ?
I think, we just need to set the requests and limits resources is they are set. Similar to this one:
https://github.com/kubeflow/katib/blob/master/sdk/python/v1beta1/kubeflow/katib/api/katib_client.py#L320-L327